static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,101 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from 'react-loader';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+
+import Jumbotron from '../components/Jumbotron';
+import {
+  startProfileEdit,
+  FETCH_PROCESSING,
+} from '../actions/index';
+import ProfileFormContainer from './ProfileFormContainer';
+import PrivacyForm from '../components/PrivacyForm';
+import ProfileProgressControls from '../components/ProfileProgressControls';
+import {
+    combineValidators,
+    privacyValidation,
+} from '../util/validation';
+
+class SettingsPage extends ProfileFormContainer {
+
+  componentDidMount() {
+    this.fetchProfile(SETTINGS.username);
+  }
+
+  componentDidUpdate() {
+    this.fetchProfile(SETTINGS.username);
+  }
+
+  componentWillMount() {
+    this.startSettingsEdit();
+  }
+
+  componentWillUnmount() {
+    this.clearProfileEdit();
+  }
+
+  startSettingsEdit() {
+    const { dispatch } = this.props;
+    dispatch(startProfileEdit(SETTINGS.username));
+  }
+
+  render() {
+    const { profiles, ui } = this.props;
+    let profile;
+    let errors;
+    let isEdit;
+    let loaded = false;
+    let username = SETTINGS.username;
+    let preferredName = SETTINGS.name;
+    
+    if (profiles[username] !== undefined) {
+      let profileFromStore = profiles[username];
+      loaded = profileFromStore.getStatus !== FETCH_PROCESSING;
+      if (profileFromStore.edit !== undefined) {
+        errors = profileFromStore.edit.errors;
+        profile = profileFromStore.edit.profile;
+        isEdit = true;
+      } else {
+        profile = profileFromStore.profile;
+        errors = {};
+        isEdit = false;
+      }
+      preferredName = profile.preferredName || SETTINGS.name;
```

Comment:
It just takes a profile object as an argument and get's the most up-to-date preferred name (`Profile.preferred_name` || `SETTINGS.preferred_name`, etc).